server/routes/api/users.ts

```diff
@@ -49,9 +59,10 @@ router.get('/activate/:activationKey', usersController.activateUser);
 // @access  Private
 router.put(
 	'/',
-	auth,
+	(req: Request, res: Response, next: NextFunction) =>
+		auth(req, res, next, true),
```

Comment:
You don't need to pass 'true' as 4th arg, because in auth middleware this param is 'true' defaulted.
Btw if we use just 'auth' rather than `(req: Request, res: Response, next: NextFunction) => auth(req, res, next)` it will shows type error?

Reply:
With only auth I'm getting server error but yea, this is a lot of types and it looks really weird for me too. I'll try to fix this in next pull requests.